src/puppetlabs/trapperkeeper/services/metrics/metrics_service.clj

```diff
@@ -7,11 +7,40 @@
 (trapperkeeper/defservice metrics-service
   metrics/MetricsService
   [[:ConfigService get-in-config]]
+
   (init [this context]
-    (core/initialize (get-in-config [:metrics] {})))
+    {:registries
+     (atom {::default-registry
+            (core/initialize (get-in-config [:metrics] {})
+                             nil)})})
+
   (stop [this context]
-    (core/stop (tk-services/service-context this)))
+    (let [{:keys [registries] :as ctx} (tk-services/service-context this)]
+      (doseq [[_ metrics-reg] @registries]
```

Comment:
If someone were to forego the service API and call into the core namespace directly, that could cause a problem, but I don't think that'd be considered a supported use case.

For downstream services using the supported API of calling into the service protocol functions to get access to the registry, we know they will be `stop`ped before this service's `stop` is called, based on how TK manages service dependencies / life cycles.  So I think this is pretty safe?